overlord/snapstate/snapmgr.go

```diff
@@ -147,6 +148,43 @@ func (snapst *SnapState) CurrentSideInfo() *snap.SideInfo {
 	panic("cannot find snapst.Current in the snapst.Sequence")
 }
 
+func (snapst *SnapState) PreviousSideInfo() *snap.SideInfo {
+	n := len(snapst.Sequence)
+	if n < 2 {
+		return nil
+	}
+	// find "current" and return the one before that
+	currentIndex := snapst.findIndex(snapst.Current)
+	if currentIndex == 0 {
+		return nil
+	}
+	return snapst.Sequence[currentIndex-1]
+}
+
+func (snapst *SnapState) findIndex(rev snap.Revision) int {
+	for i, si := range snapst.Sequence {
+		if si.Revision == rev {
+			return i
+		}
+	}
+	return -1
+}
+
+// Block returns revisions that should be blocked on refreshes,
+// computed as Sequence[currentRevisionIndex:].
+func (snapst *SnapState) Block() []snap.Revision {
+	// return revisions from Sequence[currentIndex:]
+	currentIndex := snapst.findIndex(snapst.Current)
+	if currentIndex < 0 {
+		return nil
+	}
+	out := make([]snap.Revision, 0, len(snapst.Sequence))
+	for _, si := range snapst.Sequence[currentIndex:] {
+		out = append(out, si.Revision)
+	}
+	return out
```

Comment:
These are different types, out is a []snap.Revision but snapst.Sequence is a []*snap.SideInfo.